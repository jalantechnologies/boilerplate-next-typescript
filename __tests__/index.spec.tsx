import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import HomePage from '../pages/index';

jest.mock('next/image', () => ({
  // eslint-disable-next-line @typescript-eslint/naming-convention
  __esModule: true,
  default: (props: any) => {
    // eslint-disable-next-line jsx-a11y/alt-text
    return <img {...props} />;
  },
}));

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '/',
      pathname: '',
      query: '',
      asPath: '',
    };
  },
}));

jest.mock('next-i18next', () => ({
  // this mock makes sure any components using the translate hook can use it without a warning being shown
  useTranslation: () => {
    return {
      t: (str) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

describe('With Snapshot Testing', () => {
  it('App shows Hello World heading and form', () => {
    const component = renderer.create(<HomePage />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
