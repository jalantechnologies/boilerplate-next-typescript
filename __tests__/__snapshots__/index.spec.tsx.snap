// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App shows Hello World heading and form 1`] = `
<div>
  <main>
    <div>
      <img
        alt="logo"
        height={200}
        src={
          Object {
            "default": "placeholder-file",
          }
        }
        width={200}
      />
      <div>
        <h3>
          text
        </h3>
      </div>
      <form
        action="#"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <div>
          <div
            className="input-wrapper"
          >
            <label
              htmlFor="email"
            >
              Email
            </label>
            <br />
            <input
              id="email"
              onBlur={[Function]}
              onChange={[Function]}
              type="email"
              value=""
            />
            <br />
          </div>
        </div>
        <div>
          <div
            className="input-wrapper"
          >
            <label
              htmlFor="password"
            >
              Password
            </label>
            <br />
            <input
              id="password"
              onBlur={[Function]}
              onChange={[Function]}
              type="password"
              value=""
            />
            <br />
          </div>
        </div>
        <div>
          <button
            disabled={false}
            type="submit"
          >
            Sign In
          </button>
        </div>
      </form>
    </div>
  </main>
</div>
`;
